src/arrayMethodReduce.js

```diff
@@ -5,7 +5,16 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let acc = arguments.length > 1 ? initialValue : this[0];
+    const startIndex = arguments.length > 1 ? 0 : 1;
+
+    for (let i = startIndex; i < this.length; i += 1) {
+      if ((this[i], i, this)) {
```

Comment:
This is a very unclear condition - please rewrite it in a more understandable way and explain the purpose if the condition is necessary